make-release.sh

```diff
@@ -0,0 +1,116 @@
+  esac
+  shift 1
+done
+
+usage ()
+{
+  echo "Usage: $0 --repo [GIT REPO TO EDIT] --version [VERSION TO RELEASE] [--trigger-release]"
+  echo "Example: $0 --repo git@github.com:eclipse/che-subproject --version 7.7.0 --trigger-release"; echo
+}
+
+if [[ ! ${VERSION} ]] || [[ ! ${REPO} ]]; then
+  usage
+  exit 1
+fi
+
+# derive branch from version
+BRANCH=${VERSION%.*}.x
+
+# if doing a .0 release, use master; if doing a .z release, use $BRANCH
+if [[ ${VERSION} == *".0" ]]; then
+  BASEBRANCH="master"
+else 
+  BASEBRANCH="${BRANCH}"
+fi
+
+# get sources from ${BASEBRANCH} branch
+echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
+git fetch origin "${BASEBRANCH}":"${BASEBRANCH}"
+git checkout "${BASEBRANCH}"
+
+# create new branch off ${BASEBRANCH} (or check out latest commits if branch already exists), then push to origin
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  git branch "${BRANCH}" || git checkout "${BRANCH}" && git pull origin "${BRANCH}"
+  git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}"
+  git checkout "${BRANCH}"
+fi
+
+# change VERSION file
+npm --no-git-tag-version version ${VERSION}
+
+# commit change into branch
+if [[ ${NOCOMMIT} -eq 0 ]]; then
+  COMMIT_MSG="[release] Bump to ${VERSION} in ${BRANCH}"
+  git commit -asm "${COMMIT_MSG}"
+  git pull origin "${BRANCH}"
+  git push origin "${BRANCH}"
+fi
+
+if [[ $TRIGGER_RELEASE -eq 1 ]]; then
+  # push new branch to release branch to trigger CI build
+  git fetch origin "${BRANCH}:${BRANCH}"
+  git checkout "${BRANCH}"
+  docker build -t "quay.io/mkuznets/che-dashboard:${VERSION}" -f apache.Dockerfile .
+  docker push "quay.io/mkuznets/che-dashboard:${VERSION}"
+
+  # tag the release
+  git checkout "${BRANCH}"
+  git tag "${VERSION}"
+  git push origin "${VERSION}"
+fi
+
+# now update ${BASEBRANCH} to the new snapshot version
+git fetch origin "${BASEBRANCH}":"${BASEBRANCH}"
+git checkout "${BASEBRANCH}"
+
+# change VERSION file + commit change into ${BASEBRANCH} branch
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  # bump the y digit
+  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=7.10.x, get BASE=7, NEXT=11
+  NEXTVERSION="${BASE}.${NEXT}.0-SNAPSHOT"
+else
+  # bump the z digit
+  [[ $VERSION =~ ^([0-9]+)\.([0-9]+)\.([0-9]+) ]] && BASE="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}"; NEXT="${BASH_REMATCH[3]}"; (( NEXT=NEXT+1 )) # for VERSION=7.7.1, get BASE=7.7, NEXT=2
+  NEXTVERSION="${BASE}.${NEXT}-SNAPSHOT"
+fi
+
+# change VERSION file
+npm --no-git-tag-version version ${NEXTVERSION}
+
+if [[ ${NOCOMMIT} -eq 0 ]]; then
+  BRANCH=${BASEBRANCH}
+  # commit change into branch
+  COMMIT_MSG="[release] Bump to ${NEXTVERSION} in ${BRANCH}"
+  git commit -asm "${COMMIT_MSG}"
+  git pull origin "${BRANCH}"
+
+ # PUSH_TRY="$(git push origin "${BRANCH}")"
+  # shellcheck disable=SC2181
+ # if [[ $? -gt 0 ]] || [[ $PUSH_TRY == *"protected branch hook declined"* ]]; then
```

Comment:
Are you going to uncomment/remove these code before merge?

Reply:
yes :smiley: